spec/integ/matrix-client-event-emitter.spec.js

```diff
@@ -1,24 +1,18 @@
 "use strict";
 import 'source-map-support/register';
-const sdk = require("../..");
-const HttpBackend = require("matrix-mock-request");
-const utils = require("../test-utils");
+import * as utils from "../test-utils";
+import {TestClient} from "../TestClient";
 
 describe("MatrixClient events", function() {
-    const baseUrl = "http://localhost.or.something";
     let client;
     let httpBackend;
     const selfUserId = "@alice:localhost";
     const selfAccessToken = "aseukfgwef";
 
     beforeEach(function() {
-        httpBackend = new HttpBackend();
-        sdk.request(httpBackend.requestFn);
-        client = sdk.createClient({
-            baseUrl: baseUrl,
-            userId: selfUserId,
-            accessToken: selfAccessToken,
-        });
+        const testClient = new TestClient(selfUserId, "DEVICE", selfAccessToken);
+        client = testClient.client;
+        httpBackend = testClient.httpBackend;
```

Comment:
The commit goes into some detail about why this isn't as simple as just applying the same change: https://github.com/matrix-org/matrix-js-sdk/pull/1122/commits/034b8db0704a6efc6d437062fbacc9d64c99bd75

Using ES6 imports/exports instead of the older CommonJS syntax means a slight behavioural change, and ends up breaking these tests. By using the now-fixed TestClient, things are back to a working state. Arguably we should have been using TestClient all this time, though I haven't checked the introduction dates of either files to see if that was even possible. I do suspect that the integration tests were written after many unit tests and as such have different structures.

Reply:
OK - I hope it does the right thing

Reply:
The tests pass, so hopefully that's true.